hathor/manager.py

```diff
@@ -480,16 +482,19 @@ def _initialize_components(self) -> None:
                     # TODO: deal with invalid tx
                     if tx.can_validate_full():
                         self.tx_storage.add_to_indexes(tx)
-                        assert tx.validate_full(skip_block_weight_verification=skip_block_weight_verification)
+                        assert tx.validate_full(skip_block_weight_verification=skip_block_weight_verification,
+                                                reject_locked_reward=False)
```

Comment:
Why are we adding `reject_locked_reward=False` here?

Reply:
This is a bug fix. I'm not too sure why it was passing before, but `reject_locked_reward=False` should be correct because we aren't guaranteed to have a "good" sort when initialized (a good sort is a sort that validates the block that unlocks the reward before the transaction that spends the reward). But it is actually missing a final pass on the mempool to check for transactions that could have become invalid (those are the ones where `reject_locked_reward=True` must be used).

Reply:
I'll do that, but ideally I think we should remove this code altogether possibly replacing it with two different approaches: 1) actually and explicitly rebuilding the database, 2) actually just verifying (no write operations) and only the last few blocks/transactions.

Reply:
Done. Opened #627

Reply:
Great! Thanks!